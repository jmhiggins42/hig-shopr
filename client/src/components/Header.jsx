import React from 'react';

const Header = ({ text }) => (
  <div className="row">
    <div className="col">
      {text.split('').map((ltr, i) => (
        <span key={i} className="header">
          {ltr}
        </span>
      ))}
    </div>
  </div>
);

export default Header;
