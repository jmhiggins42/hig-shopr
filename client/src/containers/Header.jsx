import React from 'react';
import EightBitBtn from '../components/EightBitBtn';

const Header = ({ isActive, btnHandler, text }) => (
  <div className={isActive ? 'row h-100' : 'row'}>
    <div className={isActive ? 'col my-auto' : ' top-bar float-left'}>
      <div className="header">
        {text.split('').map((ltr, i) => (
          <span key={i} className={isActive ? 'active' : ''}>
            {ltr}
          </span>
        ))}
      </div>
      {isActive && (
        <div className="col">
          <EightBitBtn clickHandler={btnHandler}>start</EightBitBtn>
        </div>
      )}
    </div>
  </div>
);

export default Header;
