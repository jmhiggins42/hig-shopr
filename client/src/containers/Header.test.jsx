import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header isActive={true} text="test" />, div);
    ReactDOM.unmountComponentAtNode(div);
    ReactDOM.render(<Header isActive={false} text="test" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('renders correctly', () => {
    it('active', () => {
      const tree = renderer.create(<Header isActive={true} text="test" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('inactive', () => {
      const tree = renderer.create(<Header isActive={false} text="test" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
