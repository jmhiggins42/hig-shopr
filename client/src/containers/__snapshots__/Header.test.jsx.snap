// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly active 1`] = `
<div
  className="row h-100"
>
  <div
    className="col my-auto"
  >
    <div
      className="header"
    >
      <span
        className="active"
      >
        t
      </span>
      <span
        className="active"
      >
        e
      </span>
      <span
        className="active"
      >
        s
      </span>
      <span
        className="active"
      >
        t
      </span>
    </div>
    <div
      className="col"
    >
      <button
        className="eightbit-main "
        onClick={undefined}
        type="button"
      >
        start
      </button>
    </div>
  </div>
</div>
`;

exports[`Header renders correctly inactive 1`] = `
<div
  className="row"
>
  <div
    className=" top-bar float-left"
  >
    <div
      className="header"
    >
      <span
        className=""
      >
        t
      </span>
      <span
        className=""
      >
        e
      </span>
      <span
        className=""
      >
        s
      </span>
      <span
        className=""
      >
        t
      </span>
    </div>
  </div>
</div>
`;
